### DISCUSSION TITLE
Where does the cache is stored in react-query?

### DISCUSSION THREAD

**Discussion by rajatgng** [TanStack/query#3219] [Q&A] [answered]
What I meant to ask is what is the logic behind the cache data, where does the react-query store the cache like in memory or browser storage.

I want store jwt-auth in memory, it is safe to store auth_token in react-query cache.

**Comment by TkDodo** [comment, marked as answer]
in memory

**Reply by rajatgng** [reply]
@TkDodo Thanks, so react-query stores its cache in memory.

**Reply by TkDodo** [reply]
yes

**Reply by Nparooei** [reply]
So is it safe to use? if it's in memory the other apps should not be able to intercept it, sorry if it's a dumb question, but I have a client whose penetration testing team is concerned about this. they say if you cache data  other apps can access it.

**Reply by TkDodo** [reply]
They just misinterpret what cache means. It's just a JavaScript object 🤷‍♂️

**Reply by Nparooei** [reply]
Thanks!

**Reply by TalhaZafeer** [reply]
I'm sorry what do you mean by in memory ?

**Reply by Nparooei** [reply]
It was clarified earlier it's just a JS object.

**Reply by TalhaZafeer** [reply]
@Nparooei, I get it. It might be a silly question, but I mean does this object get saved like other JS objects in memory? like person ={ age: 12, name: 'any'}

**Reply by jam6123** [reply]
@TalhaZafeer yes it's just like that same as declaring object variable

**Reply by jam6123** [reply]
@TalhaZafeer all javascript created variable are stored in memory

**Comment by saneruby** [comment]
memory on client machine or memory on server machine (where app is hosted)?

**Reply by TkDodo** [reply]
in memory

**Reply by saneruby** [reply]
is it a joke? :D memory of server or memory on client-side?

**Reply by theonlyway** [reply]
Why would it be a joke. RQ runs on the client side so why do you think it would store anything on a server?

**Reply by TkDodo** [reply]
Sorry I didn't fully grasp the question. I'm asked so often if the cache is in memory or persisted to the disc that my default answer is always "in memory".

The question itself makes little sense though. If you instantiate a queryClient on the server, it will live in memory on the server. If you do it on the client, it will be in the browser memory. So, it depends?

**Reply by saneruby** [reply]
Thank you very much for answer!
